src/simulate.jl

```diff
@@ -152,20 +149,17 @@ end
 function simulate!(
     rng::AbstractRNG, y::AbstractVector, m::LinearMixedModel{T}; β=m.β, σ=m.σ, θ=m.θ
 ) where {T}
-    length(β) == length(fixef(m)) ||
-        length(β) == length(coef(m)) ||
+    length(β) == length(m.feterm.piv) || length(β) == m.feterm.rank ||
         throw(ArgumentError("You must specify all (non-singular) βs"))
 
     β = convert(Vector{T}, β)
     σ = T(σ)
     θ = convert(Vector{T}, θ)
     isempty(θ) || setθ!(m, θ)
 
-    if length(β) ≠ length(coef(m))
-        padding = length(coef(m)) - length(β)
-        for ii in 1:padding
-            push!(β, -0.0)
-        end
+    if length(β) ≠ length(m.feterm.piv)
+        padding = length(model.feterm.piv) - m.feterm.rank
+        append!(β, fill(-0.0, padding))
```

Comment:
Also unsure how this differs functionally from what was there, which suggests to me that I'm missing something important about the lengths of these internal structures.

Reply:
doesn't differ! I assumed that a single `fill` and `append` would have better asymptotic performance than a looped `push!` (although that may not be the case depending on whether the backing memory is actually allocated -- it's generally allocated on resize to be larger than needed to avoid costs from multiple `push!`s)